src/libLocalDescriptorCache/DescDbProxy.cpp

```diff
@@ -0,0 +1,319 @@
+    {
+        grpc::protobuf::FileDescriptorProto  output;
+        m_reflectionDescDb.FindFileByName(fileName, &output);
+        std::string dbDescEntry = output.SerializeAsString();
+        host->add_file_descriptor_proto(dbDescEntry); 
+    }
+}
+
+void DescDbProxy::fetchDescNamesFromReflection()
+{
+    m_reflectionDescDb.GetServices(&m_serviceList);
+    google::protobuf::DescriptorPool descPool(&m_reflectionDescDb);
+     
+    for(const auto& serviceName: (m_serviceList)){
+        const grpc::protobuf::FileDescriptor * serviceFileDesc;
+        const grpc::protobuf::FileDescriptor * dependencyDesc;
+        const grpc::protobuf::ServiceDescriptor * serviceDesc;
+
+        // Get service file names through services in descPool
+        serviceDesc = descPool.FindServiceByName(serviceName);
+        // Catch nullptr, if services does not have a descriptor
+        if (! serviceDesc)
+        {
+            continue;
+        }
+
+        // Use filenames to retrieve service descriptors from ReflectionDB
+        serviceFileDesc = serviceDesc->file();
+
+        // Retrieve all proto files used by the service
+        int dependencyCounter = serviceFileDesc->dependency_count();
+        m_descNames.insert(serviceFileDesc->name());
+
+        //TODO: Loop only, if dependencies weren't already fetched (If parent not yet in descNames)?
+        for (int i=0; i<dependencyCounter; i++){
+            dependencyDesc = serviceFileDesc->dependency(i);
+            // Get file descriptor of imported files used in this service and search for more files
+            getDependencies(dependencyDesc);
+        }
+    }
+}
+
+void DescDbProxy::getDependencies(const grpc::protobuf::FileDescriptor * f_parentDesc)
+{       
+    std::deque<const grpc::protobuf::FileDescriptor*> todoList;
+    std::vector<const grpc::protobuf::FileDescriptor*> doneList;
+    int amountChildren;
+
+    todoList.push_front(f_parentDesc);
+
+    while (not todoList.empty())
+    {
+        if(std::find(doneList.begin(), doneList.end(),todoList.front()) != doneList.end()){
+            // Discard already processed descriptors
+            todoList.pop_front();
+        }
+        else
+        {
+            amountChildren = todoList.front()->dependency_count();
+            for (int c=0; c < amountChildren; c++)
+            {    
+                todoList.push_back(todoList.front()->dependency(c));
+            }
+            std::string currentFileName = todoList.front()->name();  
+            m_descNames.insert(currentFileName);
+            doneList.push_back(todoList.front());
+            todoList.pop_front();    
+        }          
+    }       
+}
+
+void DescDbProxy::convertHostEntryToSimpleDescDb(bool f_accessedReflectionDb, localDescDb::DescriptorDb f_dbProtoFile, std::string f_hostAddress)
+{
+    // Add services to localDB variable if DescDb on file was valid
+    std::string serviceName;
+    if (!f_accessedReflectionDb)
```

Comment:
whi is the code in this `if` scope here, this is in my opinion something which should not be part of this function.
Either move it into a separate function called by the caller (also not by this function), or restructure this flag is weird too. maybe I do not yet understand ^^